test:unit fails on main: notification-scheduler daily-reminder mock stale after #360 delivered-status change

## Bug

`npm run test:unit` has one deterministic failure on `main` (test:unit is not run by CI, so it sits unnoticed):

```
FAIL src/lib/notification-scheduler.test.ts > notification scheduler > dispatchDueNotifications sends daily reminder when window matches
AssertionError: expected +0 to be 1 — first.sent is 0
```

Reproduced on repeated runs with the frozen `now` fixture (`2026-05-29T09:02:00.000Z`), so it is not wall-clock flake. The adjacent miss-a-day test with the same frozen `now` passes.

## Root cause

PR #360 (c9f3ab5) changed the daily-reminder send path in `src/lib/notification-scheduler.ts` from fire-and-count:

```ts
await sendDailyReminderNotification({ recipientUserId, streak });
sent += 1;
```

to delivered-gated with claim rollback:

```ts
const { delivered } = await sendDailyReminderNotification({ recipientUserId, streak });
if (delivered) { sent += 1; } else { /* delete dispatch claim */ skipped += 1; }
```

(`sendDailyReminderNotification` now returns `Promise<{ delivered: boolean }>` — see the #360 review-fix commit "return delivered status from sendDailyReminderNotification".)

The unit test was not updated in #360: its `beforeEach` still stubs `sendDailyReminderNotification.mockResolvedValue(undefined)`. Destructuring `{ delivered }` from `undefined` throws a `TypeError`, which the scheduler's per-send `catch` swallows (`skipped += 1`), so `first.sent` is `0`. The miss-a-day mock was already `{ delivered: true }`, which is why that test passes.

Two secondary issues in the same area:

1. **Dead mock scaffolding:** #360 also removed `hasDispatchForLocalDate` (the only consumer of the test's `.limit()` select mock), so `queueLimitResults` / `nextLimitResult` / `limitQueryResults` / the second branch of the `dbSelect` mock are now dead code that misleads readers (it was the first suspect in diagnosing this failure).
2. **Stale-worktree pollution:** `vitest run src` (and file-path filters) substring-match test copies under `.claude/worktrees/*/src/`, so local `test:unit` runs report failures from stale checkouts that aren't part of the working tree.

## Acceptance criteria

- [ ] `sendDailyReminderNotification` test mock resolves `{ delivered: true }`, matching the real `Promise<{ delivered: boolean }>` contract
- [ ] `npx vitest run --dir src src/lib/notification-scheduler.test.ts` passes 6/6
- [ ] Full `npm run test:unit` passes from the repo root
- [ ] Dead limit-query mock scaffolding removed from the test file (no behavior change)
- [ ] `.claude/**` excluded in `vitest.config.ts` so stale worktree checkouts don't pollute local unit runs
- [ ] Scheduler source (`src/lib/notification-scheduler.ts`) unchanged — the delivered-gated behavior from #360 is intentional

## Implementation Plan

All changes are test infrastructure only — `src/lib/notification-scheduler.ts` stays untouched (the delivered-gated behavior from #360 is intentional).

**1. Fix the stale mock in `src/lib/notification-scheduler.test.ts`**
- In `beforeEach`, change `sendDailyReminderNotification.mockResolvedValue(undefined)` to `.mockResolvedValue({ delivered: true })`, matching the `Promise<{ delivered: boolean }>` contract introduced in #360 and consistent with the existing `sendMissADayNotification` mock.

**2. Remove dead limit-query scaffolding from the same test file**
- Delete `limitQueryResults`, `limitQueryIndex`, `queueLimitResults`, `nextLimitResult`, and all call sites (orphaned when #360 removed `hasDispatchForLocalDate`, the only consumer of the `.limit()` select chain).
- Simplify the `dbSelect` mock to a single shape returning `preferenceRows` — the scheduler now performs exactly one `db.select()` per run — and drop the now-unneeded `dbSelectCall` counter plus the mid-test `dbSelectCall = 0` reset.
- The second-dispatch idempotency simulation keeps working via `insertReturning.mockResolvedValueOnce([])` (claim conflict path).

**3. Exclude `.claude/**` in `vitest.config.ts`**
- Add `exclude: [...configDefaults.exclude, "**/.claude/**"]` under `test`, preserving vitest defaults, so stale worktree checkouts under `.claude/worktrees/` no longer pollute `npm run test:unit` substring-filter runs.

**Verification**
- `npx vitest run src/lib/notification-scheduler.test.ts` → 6/6 pass, and the filter no longer matches worktree copies.
- Full `npm run test:unit` from repo root → green.
- Confirm no remaining references to removed helpers (grep).

--- a/src/lib/notification-scheduler.test.ts
+++ b/src/lib/notification-scheduler.test.ts
@@ -1,9 +1,6 @@
 import { beforeEach, describe, expect, test, vi } from "vitest";
 
 let preferenceRows: Array<Record<string, unknown>> = [];
-let limitQueryResults: Array<Array<{ id: string }>> = [];
-let limitQueryIndex = 0;
-let dbSelectCall = 0;
 
 const insertReturning = vi.fn();
 const insertOnConflict = vi.fn(() => ({ returning: insertReturning }));
@@ -14,23 +11,11 @@ const dbDelete = vi.fn(() => ({ where: deleteWhere }));
 const sendDailyReminderNotification = vi.fn();
 const sendMissADayNotification = vi.fn();
 
-const dbSelect = vi.fn(() => {
-  dbSelectCall += 1;
-  if (dbSelectCall === 1) {
-    return {
-      from: vi.fn(() => ({
-        where: vi.fn(() => Promise.resolve(preferenceRows)),
-      })),
-    };
-  }
-  return {
-    from: vi.fn(() => ({
-      where: vi.fn(() => ({
-        limit: vi.fn(() => Promise.resolve(nextLimitResult())),
-      })),
-    })),
-  };
-});
+const dbSelect = vi.fn(() => ({
+  from: vi.fn(() => ({
+    where: vi.fn(() => Promise.resolve(preferenceRows)),
+  })),
+}));
 
 vi.mock("@/db", () => ({
   db: {
@@ -78,27 +63,14 @@ const basePrefs = {
   missADayEnabled: false,
 };
 
-function queueLimitResults(...results: Array<Array<{ id: string }>>) {
-  limitQueryResults = results;
-  limitQueryIndex = 0;
-}
-
-function nextLimitResult(): Array<{ id: string }> {
-  const result = limitQueryResults[limitQueryIndex] ?? [];
-  limitQueryIndex += 1;
-  return result;
-}
-
 describe("notification scheduler", () => {
   beforeEach(() => {
     vi.clearAllMocks();
     vi.resetModules();
-    dbSelectCall = 0;
     preferenceRows = [basePrefs];
-    queueLimitResults([]);
     sendDailyReminderNotification.mockReset();
     sendMissADayNotification.mockReset();
-    sendDailyReminderNotification.mockResolvedValue(undefined);
+    sendDailyReminderNotification.mockResolvedValue({ delivered: true });
     sendMissADayNotification.mockResolvedValue({ delivered: true });
     insertReturning.mockReset();
     insertReturning.mockResolvedValue([{ id: "dispatch-1" }]);
@@ -128,14 +100,10 @@ describe("notification scheduler", () => {
   });
 
   test("dispatchDueNotifications sends daily reminder when window matches", async () => {
-    queueLimitResults([]);
-
     const { dispatchDueNotifications } = await import("./notification-scheduler");
     const now = new Date("2026-05-29T09:02:00.000Z");
 
     const first = await dispatchDueNotifications(now);
-    dbSelectCall = 0;
-    queueLimitResults([{ id: "existing-dispatch" }]);
     insertReturning.mockResolvedValueOnce([]);
     const second = await dispatchDueNotifications(now);
 
@@ -146,7 +114,6 @@ describe("notification scheduler", () => {
 
   test("dispatchDueNotifications sends miss-a-day when yesterday was missed", async () => {
     preferenceRows = [{ ...basePrefs, missADayEnabled: true, dailyReminderEnabled: false }];
-    queueLimitResults([]);
     userCompletedSessionOnDate.mockResolvedValue(false);
 
     const { dispatchDueNotifications } = await import("./notification-scheduler");
@@ -159,7 +126,6 @@ describe("notification scheduler", () => {
 
   test("dispatchDueNotifications skips miss-a-day when user meditated today", async () => {
     preferenceRows = [{ ...basePrefs, missADayEnabled: true, dailyReminderEnabled: false }];
-    queueLimitResults([]);
     userCompletedSessionOnDate.mockImplementation(async (_userId, date) => date === "2026-05-29");
 
     const { dispatchDueNotifications } = await import("./notification-scheduler");

--- a/vitest.config.ts
+++ b/vitest.config.ts
@@ -1,5 +1,5 @@
 import { fileURLToPath } from "node:url";
-import { defineConfig } from "vitest/config";
+import { configDefaults, defineConfig } from "vitest/config";
 
 export default defineConfig({
   resolve: {
@@ -9,5 +9,7 @@ export default defineConfig({
   },
   test: {
     environment: "node",
+    // keep stale agent worktree checkouts out of local runs
+    exclude: [...configDefaults.exclude, "**/.claude/**"],
   },
 });

